### ISSUE TITLE
TEAMS: Pre-Training questions

### ISSUE THREAD

**Issue by stefan-it** [tensorflow/models#10567]
Hi,

I have some questions about the "Training ELECTRA Augmented with Multi-word Selection" (TEAMS) implementation:

At the moment, there's no section about generating pre-training dataset. I would really like to know if the `create_pretraining.py` script from [here](official/nlp/data/create_pretraining_data.py) can be used (although, ELECTRA has a different masking strategy and e.g. `dupe_factor` could be 1?!). If not, how is it possible to create TFRecords for the TEAMS approach?

The pre-training procedure is currently not documented. Looking at another project (e.g. Token Dropping) it seems that `train.py` is missing. So I think one could copy it and simply change the experiment config (see [here](https://github.com/tensorflow/models/blob/master/official/projects/token_dropping/train.py#L27)).

I'm interesting in pre-training and convert TEAMS model weights to be compatible with Transformers library, so any help regarding to the pre-training data generation would highly be appreciated!

**Comment by stefan-it** [comment]
I think the pre-training data from `create_pretraining.py` can be used:

https://github.com/tensorflow/models/blob/af1a6c576671e9e4f4f58be6bfe2fcb513546322/official/projects/teams/teams_experiments.py#L55-L69

**Comment by stefan-it** [comment]
TEAMS paper mentions a dupe factor of 40:

![image](https://user-images.githubusercontent.com/20651387/160498526-6aa1c39c-078f-4623-8843-02d8798bef43.png)

**Comment by saberkun** [comment]
@remenberl @jmshen1994

**Comment by saberkun** [comment]
I will add the train.py which should be trivial. For training data generation and process, please check Jialu and Jiaming as the original paper authors.

**Comment by remenberl** [comment]
As you mentioned, you can use default script for generating pretraining dataset. https://github.com/tensorflow/models/blob/master/official/nlp/data/create_pretraining_data.py

We use a similar script for data preparation.

**Comment by saberkun** [comment]
Added train.py as https://github.com/tensorflow/models/commit/7df0c6f7b6f021828769a96054e85909da01fd05. I tested internally

**Comment by stefan-it** [comment]
Hi @saberkun , thanks for that! I can confirm that pre-training is now working on my own dataset :hugs:
